lib/models/package-info-cache/package-info.js

```diff
@@ -346,21 +346,25 @@ class PackageInfo {
     if (invalidPackages.length > 0) {
       let typeName = this.project ? 'project' : 'addon';
 
-      logger.info('');
-      logger.info(`The 'package.json' file for the ${typeName} at ${this.realPath}`);
+      let msg = `The 'package.json' file for the ${typeName} at ${this.realPath}`;
 
       let relativePath;
-
       if (invalidPackages.length === 1) {
         relativePath = path.relative(this.realPath, invalidPackages[0].realPath);
-        logger.info(`  specifies an invalid, malformed or missing addon at relative path '${relativePath}'`);
+        msg = `${msg}\n  specifies an invalid, malformed or missing addon at relative path '${relativePath}'`;
       } else {
-        logger.info('  specifies invalid, malformed or missing addons at relative paths');
+        msg = `${msg}\n  specifies invalid, malformed or missing addons at relative paths`;
         invalidPackages.forEach(packageInfo => {
           let relativePath = path.relative(this.realPath, packageInfo.realPath);
-          logger.info(`    '${relativePath}'`);
+          msg = `${msg}\n    '${relativePath}'`;
         });
       }
+
+      if (process.env.ERROR_ON_INVALID_ADDON) {
```

Comment:
Let's prefix this with `EMBER_CLI_`:

```suggestion
      if (process.env.EMBER_CLI_ERROR_ON_INVALID_ADDON) {
```